From: Debasish Biswas <debasishbsws.dev@gmail.com>

This patch is required to update k8s.io/kubernetes to v1.32.10. In this latest version, k8s.io/kubernetes/pkg/volume/util.GenerateVolumeName has been removed, and upstream has shifted to using a simpler implementation instead.
Reference:
- https://github.com/kubernetes-sigs/blob-csi-driver/commit/10b9ead42c8abcc787742021dc7a0e3d20b45534#diff-a31cbcaeb4df90a0965ae085c1733780986409b34f03a3d7e3ea8cd09de30e1e
- https://github.com/kubernetes-sigs/blob-csi-driver/commit/20e06b7a4b2057a6f7039614ffd7b233759f7471
We can't cherry-pick this commit because of conflicting changes in go.sum, so we're creating this patch to incorporate only the changes we need.

diff --git a/pkg/blob/blob.go b/pkg/blob/blob.go
index 2b38462c4..af9cd0655 100644
--- a/pkg/blob/blob.go
+++ b/pkg/blob/blob.go
@@ -38,7 +38,6 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/klog/v2"
-	k8sutil "k8s.io/kubernetes/pkg/volume/util"
 	mount "k8s.io/mount-utils"
 	utilexec "k8s.io/utils/exec"
 
@@ -425,7 +424,7 @@ func getValidContainerName(volumeName, protocol string) string {
 	if !checkContainerNameBeginAndEnd(containerName) || len(containerName) < containerNameMinLength {
 		// now we set as 63 for maximum container name length
 		// todo: get cluster name
-		containerName = k8sutil.GenerateVolumeName(fmt.Sprintf("pvc-%s", protocol), uuid.NewUUID().String(), 63)
+		containerName = generateVolumeName(fmt.Sprintf("pvc-%s", protocol), uuid.NewUUID().String(), 63)
 		klog.Warningf("requested volume name (%s) is invalid, regenerated as (%q)", volumeName, containerName)
 	}
 	return strings.Replace(containerName, "--", "-", -1)
@@ -1111,3 +1110,18 @@ func isReadOnlyFromCapability(vc *csi.VolumeCapability) bool {
 	return (mode == csi.VolumeCapability_AccessMode_MULTI_NODE_READER_ONLY ||
 		mode == csi.VolumeCapability_AccessMode_SINGLE_NODE_READER_ONLY)
 }
+
+// generateVolumeName returns a PV name with clusterName prefix. The function
+// should be used to generate a name of GCE PD or Cinder volume. It basically
+// adds "<clusterName>-dynamic-" before the PV name, making sure the resulting
+// string fits given length and cuts "dynamic" if not.
+func generateVolumeName(clusterName, pvName string, maxLength int) string {
+	prefix := clusterName + "-dynamic"
+	pvLen := len(pvName)
+	// cut the "<clusterName>-dynamic" to fit full pvName into maxLength
+	// +1 for the '-' dash
+	if pvLen+1+len(prefix) > maxLength {
+		prefix = prefix[:maxLength-pvLen-1]
+	}
+	return prefix + "-" + pvName
+}
diff --git a/pkg/blob/controllerserver.go b/pkg/blob/controllerserver.go
index 2bf3491e1..117aea34d 100644
--- a/pkg/blob/controllerserver.go
+++ b/pkg/blob/controllerserver.go
@@ -610,6 +610,11 @@ func (d *Driver) ValidateVolumeCapabilities(ctx context.Context, req *csi.Valida
 	}, nil
 }
 
+// ControllerModifyVolume modify volume
+func (d *Driver) ControllerModifyVolume(_ context.Context, _ *csi.ControllerModifyVolumeRequest) (*csi.ControllerModifyVolumeResponse, error) {
+	return nil, status.Error(codes.Unimplemented, "")
+}
+
 func (d *Driver) ControllerPublishVolume(_ context.Context, _ *csi.ControllerPublishVolumeRequest) (*csi.ControllerPublishVolumeResponse, error) {
 	return nil, status.Error(codes.Unimplemented, "ControllerPublishVolume is not yet implemented")
 }
